docs/switch.rst

```diff
@@ -0,0 +1,369 @@
+           # Perform GDAL operations in this context.
+           # ...
+           # Done.
+
+   # At this point, configuration options are set back to their
+   # previous (possibly unset) values. The raster block cache size
+   # is returned to its default (5% of available RAM) and debugging
+   # messages are disabled.
+
+As mentioned previously, ``gdal`` has no such abstraction for the GDAL
+environment. The nearest approximation would be something like the code
+below.
+
+.. code-block:: python
+
+   from osgeo import gdal
+
+   # Define a new configuration, save the previous configuration,
+   # and then apply the new one.
+   new_config = {
+       'CPL_DEBUG': 'ON', 'GDAL_CACHEMAX': '512'}
+   prev_config = {
+       key: gdal.GetConfigOption(key) for key in new_config.keys()}
+   for key, val in new_config.items():
+       gdal.SetConfigOption(key, val)
+
+   # Perform GDAL operations in this context.
+   # ...
+   # Done.
+
+   # Restore previous configuration.
+   for key, val in prev_config.items():
+       gdal.SetConfigOption(key, val)
+
+Rasterio provides this with a single Python statement.
+
+.. code-block:: python
+
+   with rasterio.Env(CPL_DEBUG=True, GDAL_CACHEMAX=512):
+       # ...
+
+
+Format Drivers
+==============
+
+``gdal`` provides objects for each of the GDAL format drivers. With Rasterio,
+format drivers are represented by strings and are used only as arguments to
+functions like ``rasterio.open()``.
+
+.. code-block:: python
+
+   dst = rasterio.open('new.tif', 'w', format='GTiff', **kwargs)
+
+Rasterio uses the same format driver names as GDAL does.
+
+Datasets
+========
+
+Rasterio uses URIs to identify datasets, with schemes for different protocols.
+The GDAL bindings have their own special syntax.
+
+Unix-style filenames such as ``/var/data/example.tif`` identify dataset files
+for both Rasterio and ``gdal``. Rasterio also accepts 'file' scheme URIs
+like ``file:///var/data/example.tif``.
+
+Rasterio identifies datasets within ZIP or tar archives using Apache VFS style
+identifiers like ``zip:///var/data/example.zip!example.tif`` or
+``tar:///var/data/example.tar!example.tif``.
+
+Datasets served via HTTPS are identified using 'https' URIs like
+``https://landsat-pds.s3.amazonaws.com/L8/139/045/LC81390452014295LGN00/LC81390452014295LGN00_B1.TIF``.
+
+Datasets on AWS S3 are identified using 's3' scheme identifiers like
+``s3://landsat-pds/L8/139/045/LC81390452014295LGN00/LC81390452014295LGN00_B1.TIF``.
+
+With ``gdal``, the equivalent identifiers are respectively
+``/vsizip//var/data/example.zip/example.tif``,
+``/vsitar//var/data/example.tar/example.tif``,
+``/vsicurl/landsat-pds.s3.amazonaws.com/L8/139/045/LC81390452014295LGN00/LC81390452014295LGN00_B1.TIF``,
+and
+``/vsis3/landsat-pds/L8/139/045/LC81390452014295LGN00/LC81390452014295LGN00_B1.TIF``.
+
+To help developers switch, Rasterio will accept these identifiers, too, and
+dispatch them to the proper format drivers and protocol implementations.
```

Comment:
Currently we just silently accept those alternate identifiers without warning - maybe we need to start formally raising a DeprecationWarning (off topic)